storage/memory/peer_store.go

```diff
@@ -96,6 +97,25 @@ func New(cfg Config) (storage.PeerStore, error) {
 		}
 	}()
 
+	ps.wg.Add(1)
+	go func() {
+		defer ps.wg.Done()
+		t := time.NewTicker(1 * time.Second)
+		for {
+			select {
+			case <-ps.closing:
+				t.Stop()
+				select {
+				case <-t.C:
+				default:
+				}
+				return
+			case now := <-t.C:
+				ps.setClock(now.UnixNano())
```

Comment:
`UnixNano` doesn't cause another syscall. `t.C` already gives us the current time, so we might as well use it :)